### ISSUE TITLE
doasedit ignores mktemp, cp errors and can't handle filenames starting with -

### ISSUE THREAD

**Issue by oguz-ismail** [slicer69/doas#127]
See:
```
$ echo foo >-bar.txt
$ doasedit -- -bar.txt
File does not exist or is a special file/link.
$ doasedit -bar.txt
cp: invalid option -- '.'
Try 'cp --help' for more information.
cmp: invalid option -- 'a'
cmp: Try 'cmp --help' for more information.
doas: not installed setuid
cmp: invalid option -- 'a'
cmp: Try 'cmp --help' for more information.
Copying file back to -bar.txt failed. Press Ctrl-C to abort or Enter to try again.
^C
```
The second `doasedit` command opens `vi` but doesn't show the contents of `-bar.txt`.

```
$ touch x.txt
$ TMPDIR=/ VISUAL=ls doasedit x.txt
mktemp: failed to create file via template ‘/doasedit.XXXXXXXX’: Permission denied
cp: cannot create regular file '': No such file or directory
ls: cannot access '': No such file or directory
```

**Comment by slicer69** [comment]
The first issue, where doasedit doesn't work with filenames with a leading dash (like -bar.txt) has been fixed. It's in the repository now, under the master branch, and will be in the next release.

In the second example,  what we see there is (mostly) expected behaviour because the temp directory has been set to an area where the user doesn't have write access. However, the error doesn't make it clear why the command is failing. I've fixed this so doasedit exits gracefully.
